packages/react-core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -272,8 +273,23 @@ const DatePickerBase = (
           </InputGroup>
         </div>
       </Popover>
-      {helperText && <div className={styles.datePickerHelperText}>{helperText}</div>}
-      {errorText.trim() && <div className={css(styles.datePickerHelperText)}>{errorText}</div>}
+      {helperText && React.isValidElement(helperText) && (
```

Comment:
So this API is quite different from the Progress component.  I think we should try and be consistent.  This approach allow the consumer to pass in a string and then we generate the Helper text for them.  THe progress the consumer always has to pass it.

We can either go with what you have here or what progress already does.  Either way, we should also update the prop comment to be a little more explicit about what the prop expects.

@wise-king-sullyman I am interested on your thoughts here.

Reply:
Yeah I can definitely see the consumer benefits in something that is more automated like this when possible. This is actually a bit more like my original proposal for the implementation in Progress.

I can also see the benefits for maintainability and API simplicity with the approach of just making the consumer pass in the entirely of the helper text though.

I would probably lean _slightly_ towards just making the consumer put the helper text together, and maybe having an example/demo that shows how we suggest they put it together in this case. Only slightly though. I wonder if there's any good way for us to get data about how HelperText is being used in progress, as if it's largely just basic implementations that would probably nudge me in the other direction.

Also here's a link to my PR adding helper text to Progress for reference to the discussion there: https://github.com/patternfly/patternfly-react/pull/8307

Also also I wouldn't block for it, but I'd love to see some tests for this new functionality, especially if we land on this implementation with more internal logic.

Reply:
I had initially preserved how Datepicker was building helper text (wrapping it internally) but added the top use case when I thought about all the other props that weren't being exposed to the user (like `icon` or other `variants` of `HelperText`). I can remove the `string` case handling though, just LMK.

Reply:
Our DatePicker example uses a `string` literal for instance, but I can update it to pass HelperText instead.

Reply:
Yes,  Can you update the example.